java/core/src/java/org/apache/orc/impl/WriterImpl.java

```diff
@@ -189,8 +189,12 @@ public WriterImpl(FileSystem fs,
       this.rowIndexStride = 0;
     }
 
-    // ORC-1343: We ignore `opts.isBuildIndex` due to the lack of reader support
-    this.buildIndex = rowIndexStride > 0;
+    if (!opts.isBuildIndex() && rowIndexStride > 0) {
+      throw new IllegalArgumentException("Please set rowIndexStride to zero " +
```

Comment:
Or we could set the rowIndexStride=0 directly.

Reply:
So, based on this pr, it is not possible to write buildIndex = false and rowIndexStride > 0 , right?

Reply:
Yes, you are right.

Reply:
I was looking at the WriterOptions. Was thinking if we should integrate if there as well. There seem to be 2 places of interest right now.

Constructor. We currently have
```java
      buildIndex = OrcConf.ENABLE_INDEXES.getBoolean(tableProperties, conf);
      rowIndexStrideValue =
          (int) OrcConf.ROW_INDEX_STRIDE.getLong(tableProperties, conf);
```
We should set rowIndexStrideValue to 0 if `buildIndex` is false
 
Setter for rowIndexStride (I don't see a setter for buildIndex)
```java
    public WriterOptions rowIndexStride(int value) {
      rowIndexStrideValue = value;
      return this;
    }
```
Adapt this to say if rowIndexStride is 0 then set buildIndex false, else if rowIndexStride > 0 set buildIndex true.

Reply:
@pavibhai Good suggesions, thank you, let me try.

Reply:
Now we have two options to support the configuration:
1. Throw an exception directly when the configuration mismatch.
2. Help users configure `orc.row.index.stride=0` when `orc.create.index=false` is configured, 

Using the first scheme will allow users to strongly perceive the orc.row.index.stride parameter, but it will increase the difficulty of use. Users need to manually set the orc.row.index.stride parameter.

Reply:
My preference would be for automatic determination so 2.

* if `orc.row.index.stride = 0` then set `orc.create.index=false`
* if `orc.row.index.stride > 0` then set `orc.create.index=true`

Reply:
Thank you . Updated it.